src/random.cpp

```diff
@@ -467,6 +467,13 @@ static void SeedEvents(CSHA512& hasher)
     events_hasher.Write(events_hash, 32);
 }
 
+// buffer[32] is intentionally not initialized, see https://github.com/bitcoin/bitcoin/pull/17627
+// Suppress memory sanitizer because it can't see that syscall initializes this in GetOSRand().
+#if defined(__has_feature) && defined(HAVE_SYS_GETRANDOM)
+#  if __has_feature(memory_sanitizer)
+__attribute__((no_sanitize("memory")))
+#  endif
+#endif
```

Comment:
This is a clang bug, right? Should link to the commit that fixed it. And since you are compiling clang from source you might as well use that commit and then remove this code blob here.

Reply:
The memory sanitizer appears to be a Google project, and they don't except `syscall` support to be fixed anytime soon. Only a more narrow case of `getrandom` was fixed, which is of no use to us: https://github.com/google/sanitizers/issues/852#issuecomment-327305535

Reply:
Ok, fine. I'd still prefer a suppression in a suppressions file if possible. Modifying the code like this makes it harder to read and potentially impossible to compile on some environments.

Reply:
@practicalswift any idea how to achieve such a suppressions file?
The outer `#if defined(__has_feature)` is there to make sure other environments don't trip over it. I could put it in a macro, or we can do that if we ever need it again.

Reply:
@Sjors Sure! :) You need to use the sanitizer special case list functionality by passing `-fsanitize-blacklist=test/sanitizer_suppressions/msan` as part of `CFLAGS`/`CXXFLAGS`. You need to create the file `test/sanitizer_suppressions/msan` following the file format specified here: https://clang.llvm.org/docs/SanitizerSpecialCaseList.html#format

Reply:
I prefer to punt on this until we need it in more places...

Reply:
> I prefer to punt on this until we need it in more places...

So this is the only place where MSAN special treatment is needed? That's pretty good, I had assumed it would be the first of many.

Reply:
I don't know yet. The binary exits once it finds a single problem. The next problem in line is in a different file, but who know what happens after that.